packages/ckeditor5-ui/src/dialog/dialogview.ts

```diff
@@ -0,0 +1,635 @@
+/**
+ * @license Copyright (c) 2003-2023, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/**
+ * @module ui/dialog/dialogview
+ */
+
+import {
+	type EventInfo,
+	type Locale,
+	type CollectionChangeEvent,
+	type DecoratedMethodEvent,
+	KeystrokeHandler,
+	FocusTracker,
+	Rect,
+	global,
+	toUnit
+} from '@ckeditor/ckeditor5-utils';
+import ViewCollection from '../viewcollection';
+import View from '../view';
+import FormHeaderView from '../formheader/formheaderview';
+import ButtonView from '../button/buttonview';
+import FocusCycler, { isViewWithFocusCycler, type FocusCyclerBackwardCycleEvent, type FocusCyclerForwardCycleEvent } from '../focuscycler';
+import DraggableViewMixin, { type DraggableView, type DraggableViewDragEvent } from '../bindings/draggableviewmixin';
+import DialogActionsView, { type DialogActionButtonDefinition } from './dialogactionsview';
+
+// @if CK_DEBUG_DIALOG // const RectDrawer = require( '@ckeditor/ckeditor5-utils/tests/_utils/rectdrawer' ).default;
+
+import cancelIcon from '@ckeditor/ckeditor5-core/theme/icons/cancel.svg';
+import type EditorUI from '../editorui/editorui';
+
+export enum DialogViewPosition {
+	SCREEN_CENTER = 'screen-center',
+	CURRENT_ROOT_CENTER = 'current-root-center',
+	CURRENT_ROOT_NE = 'current-root-ne',
+	CURRENT_ROOT_NW = 'current-root-nw'
+}
```

Comment:
I implemented the requested positions. As for the `ABOVE` and `BELOW`, there's already the partial logic responsible for keeping the dialog visible within `DialogView#moveTo()` method. It's missing just the bottom edge part. Maybe finishing it would be enough? See how it works for the `ABOVE` position in the first demo.